### DISCUSSION TITLE
problem with replace_datafile

### DISCUSSION THREAD

**Discussion by albenard** [gdcc/pyDataverse#170] [Q&A]
I use pydataverse and it’s very nice but i have a problem when i use api.replace_datafile. I describe the problem :
               In all case (upload or replace) i have this code :
                  df = DataFile()
                  df.set(
                      {"pid": doi_ds,"filename": chemin_fichier,"categories":["Data"],"forceReplace":"true"})

1.	I upload a file with this line : resultat_upload = api.upload_datafile(doi_ds, df.filename, df.json())
        It’s ok and Dataverse detect the type of file (for example MS Word for a docx file)
2.	I would to replace the file with this code :  resultat_remplace = api.replace_datafile(identifier=pid_fichier, filename=chemin_fichier,
                                         json_str=df.json(), is_filepid=True)
        I have an error message : b'{"status":"ERROR","message":"The original file (MS Word) and replacement file (Plain Text) are different file types."}'
        It’s the same thing with other extension file (for example .xyz) when Dataverse is able to fix a type of file (for view tools i suppose).

api.replace_datafile don’t fix the content-type and Dataverse make the job but for replacing there is a bug. I understand in the code of dataverse that i can force but "forceReplace" :"true" has no effect. With native API in curl there is no problem.
Can you help me.
Thanks for advance.
I can give some detail if necessary or make test.

**Comment by JR-1991** [comment]
Hi @albenard! Thanks for submission your issue 🙌

Would you mind submitting this issue in the `Issues` tab? It would really help others who may be experiencing the same problem. I'll do my best to reproduce the issue in the meantime!

**Comment by pdurbin** [comment]
It looks like @albenard created this issue:

- #171

Thanks!
